### PR TITLE
Improve exception handling of heron-explorer.

### LINKED ISSUE #1117: suppress explorer's traceback when tracker is not running

### PR DESCRIPTION
**Pull request by objmagic**
1. unify format of error message when exception is raised
2. fix two places where exception will not be caught and traceback will be printed out.  
    This is ugly so I fixed it. (resolves #1117)

### CODE REVIEW

File: heron/common/src/python/utils.py [hunk 1]
```diff
@@ -353,7 +353,7 @@ def get_clusters():
   try:
     return instance.run_sync(lambda: API.get_clusters())
   except Exception as ex:
-    Log.error('Error: %s' % str(ex))
+    Log.error(str(ex))
```
Comment (kramasamy) [inline heron/common/src/python/utils.py]: this will print the exception stack.

File: heron/common/src/python/utils.py [hunk 2]
```diff
@@ -364,7 +364,7 @@ def get_logical_plan(cluster, env, topology, role):
   try:
     return instance.run_sync(lambda: API.get_logical_plan(cluster, env, topology, role))
   except Exception as ex:
-    Log.error('Error: %s' % str(ex))
+    Log.error(str(ex))
```
Comment (kramasamy) [inline heron/common/src/python/utils.py]: same as before.

File: heron/common/src/python/utils.py [hunk 3]
```diff
@@ -377,6 +377,7 @@ def get_topology_info(*args):
     return instance.run_sync(lambda: API.get_topology_info(*args))
   except Exception as ex:
     Log.error(str(ex))
+    Log.error('Failed to get topology info')
```
Comment (kramasamy) [inline heron/common/src/python/utils.py]: same as before.

File: heron/common/src/python/utils.py [hunk 4]
```diff
@@ -387,7 +388,7 @@ def get_topology_metrics(*args):
     return instance.run_sync(lambda: API.get_comp_metrics(*args))
   except Exception as ex:
     Log.error(str(ex))
-    Log.error("Failed to retrive metrics of component \'%s\'" % args[3])
+    Log.error("Failed to retrive metrics of topology \'%s\'" % args[3])
```
Comment (kramasamy) [inline heron/common/src/python/utils.py]: same as before.

File: heron/common/src/python/utils.py [hunk 5]
```diff
@@ -398,7 +399,10 @@ def get_component_metrics(component, cluster, env, topology, role):
     result = get_topology_metrics(
         cluster, env, topology, component, [], all_queries, [0, -1], role)
     return result["metrics"]
-  except:
+  except Exception as ex:
+    Log.error(str(ex))
```
Comment (kramasamy) [inline heron/common/src/python/utils.py]: same as before.

File: heron/common/src/python/utils.py (hunk 1 above)
Comment (objmagic) [inline heron/common/src/python/utils.py]: For a Python program ``test.py``:
````
def a():
  return 1 / 0

def b():
  return a()

def c():
  return b()

try:
  c()
except Exception as ex:
  print(str(ex))
  pass
````

``python test.py`` will only print out "division by zero"

An informative error will be printed out after ``Log.error(str(ex))`` (I can add cluster name, though)

Comment (objmagic) [inline heron/common/src/python/utils.py]: If you print out exception using ``print(ex)``, you will see stack trace.

**Comment by kramasamy** [comment]
@objmagic - are we adding a verbose option? or it is going to be in a later PR?

**Comment by objmagic** [comment]
It already has a verbose option.

**Comment by kramasamy** [comment]
👍

**Pull request merged**